### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
@jpkrohling An initial example that works locally - ignore the README.md files in the servicea/b folders for now - the top level README.md should provide all the info required to try it out.

Unfortunately currently requires the java-agent and a PR for the web-servlet-filter to be built - so relies on snapshots. Might make it difficult to try in Openshift currently.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+NOTE: Once the OpenTracing Agent has been released, the Hawkular APM project will be updated to mark the
+`HttpURLConnection` directly with this request property.
+
+### Run the instrumented example
+
+Before being able to try the OpenTracing Java Agent, it will be necessary to obtain the agent jar.
+
+It will then be necessary to [startup a Hawkular APM server](https://hawkular.gitbooks.io/hawkular-apm-user-guide/content/quickstart/) and set up the `HAWKULAR_APM_USER`, `HAWKULAR_APM_PASSWORD` and `HAWKULAR_APM_URI` environment
```
Comment (jpkrohling) [inline README.md]: s/`HAWKULAR_APM_USER`/`HAWKULAR_APM_USERNAME`/

File: README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+`opentracing-agent/target` folder.
+
+NOTE: This example is currently dependent upon [this PR](https://github.com/opentracing-contrib/java-web-servlet-filter/pull/11) therefore this version will need to be cloned and built locally.
+
+## The Example
+
+The example services are based on [this example](http://github.com/obsidian-toaster/quick_rest_springboot-tomcat),
+but modified to have two spring boot services, one calling the other.
+
+The first service (Service A) is called by an external client (e.g. curl) using the `/greeting' endpoint, supplying
```
Comment (jpkrohling) [inline README.md]: Fix the ' after `/greeting`.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+{"id":2,"content":"Hello, Fred!"}
+```
+
+## OpenTracing Java Agent Instrumentation
+
+### Instrument the technologies and frameworks
+
+The OpenTracing framework integrations and rules are added to the services as dependencies, e.g.
+
+```xml
```
Comment (jpkrohling) [inline README.md:98]: I think it doesn't belong here. At a first sight, it looks like I have to include these somewhere, which isn't the case.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+In these example services, we will use the
+[servlet integration](https://github.com/opentracing-contrib/java-web-servlet-filter) and the direct
+instrumentation rules for the Java `HttpURLConnection`.
+
+NOTE: Currently the agent instrumentation rules for installing the servlet filter are provided as a
+separate maven dependency. Eventually these rules will be bundled with the servlet integration project.
+
+### Add an OpenTracing Compliant Tracer
```
Comment (jpkrohling) [inline README.md:126]: This isn't required either, as it's there already.

File: README.md (hunk 3 above)
Comment (jpkrohling) [reply README.md:98]: Perhaps add a section at the end called "How it works", to explain how it all fits together?

File: servicea/src/main/java/com/example/servicea/Greeting.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (jpkrohling) [inline servicea/src/main/java/com/example/servicea/Greeting.java:2]: Is this license header accurate?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+Then re-run the client:
+
+```
+curl http://localhost:8080/greeting -d name=Fred
+```
+
+Finally, start up the Hawkular APM UI and you should see the interactions between the two services on the
+Distributed Tracing page.
```
Comment (jpkrohling) [inline README.md]: Could you add a screenshot with the expected outcome? I got this and I'm not sure I'm supposed to see more things:

image

image

File: servicea/src/main/java/com/example/servicea/Greeting.java (hunk 5 above)
Comment (objectiser) [reply servicea/src/main/java/com/example/servicea/Greeting.java:2]: This was the license used on the original obsidian example upon which this is based.

File: README.md (hunk 4 above)
Comment (objectiser) [reply README.md:126]: Moved to "How it works" section.

**Comment by objectiser** [comment]
@jpkrohling Thanks! Hopefully all of your suggestions have been dealt with - let me know if there is anything wrong.

**Comment by jpkrohling** [comment]
LGTM!

**Comment by objectiser** [comment]
@jpkrohling Thanks for the review.

**Pull request merged**